app/models/registration.rb

```diff
@@ -521,4 +521,56 @@ def competition_events_changed?
   def serializable_hash(options = nil)
     super(DEFAULT_SERIALIZE_OPTIONS.merge(options || {}))
   end
+
+  def attempt_auto_accept
+    return false if Rails.env.production? && EnvConfig.WCA_LIVE_SITE?
+
+    failure_reason = auto_accept_failure_reason
+    if failure_reason.present?
+      log_auto_accept_failure(failure_reason)
+      return false
+    end
+
+    update_payload = build_auto_accept_payload
+    auto_accepted_registration = Registrations::RegistrationChecker.apply_payload(self, update_payload)
+
+    if auto_accepted_registration.valid?
+      update_lanes!(
+        update_payload,
+        'Auto-accept',
+      )
+      true
+    else
+      log_auto_accept_failure(auto_accepted_registration.errors.messages.values.flatten)
+      false
+    end
+  end
+
+  private def auto_accept_failure_reason
+    return 'Competitor still has outstanding registration fees' if outstanding_entry_fees > 0
+    return 'Auto-accept is not enabled for this competition.' unless competition.auto_accept_registrations?
```

Comment:
How would the code even reach this point? `auto_accept_failure_reason` is only called when auto accept is attempted. And auto accept is only attempted if it's turned on, or not?

Reply:
As discussed in Huddle, added a check in RegistrationPayment, but leaving this here for redundancy